src/backend/distributed/commands/utility_hook.c

```diff
@@ -731,12 +732,32 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 		}
 		PG_CATCH();
 		{
-			ereport(ERROR,
-					(errmsg("CONCURRENTLY-enabled index command failed"),
-					 errdetail("CONCURRENTLY-enabled index commands can fail partially, "
-							   "leaving behind an INVALID index."),
-					 errhint("Use DROP INDEX CONCURRENTLY IF EXISTS to remove the "
-							 "invalid index, then retry the original command.")));
+			MemoryContextSwitchTo(savedContext);
```

Comment:
Maybe add this comment:

/* CopyErrorData() requires (CurrentMemoryContext != ErrorContext) */